apps/extension/src/storage/LocalStorage.ts

```diff
@@ -0,0 +1,188 @@
+import { KVStore } from "@namada/storage";
+import * as E from "fp-ts/Either";
+import * as t from "io-ts";
+import { ExtStorage } from "./Storage";
+
+enum BridgeType {
+  IBC = "ibc",
+  Ethereum = "ethereum-bridge",
+}
+
+const Chain = t.intersection([
+  t.type({
+    alias: t.string,
+    bech32Prefix: t.string,
+    bip44: t.type({
+      coinType: t.number,
+    }),
+    bridgeType: t.array(
+      t.keyof({ [BridgeType.IBC]: null, [BridgeType.Ethereum]: null })
+    ),
+    chainId: t.string,
+    currency: t.intersection([
+      t.type({
+        symbol: t.string,
+        token: t.string,
+      }),
+      t.partial({
+        address: t.string,
+        gasPriceStep: t.type({
+          average: t.number,
+          high: t.number,
+          low: t.number,
+        }),
+      }),
+    ]),
+    extension: t.type({
+      alias: t.string,
+      id: t.keyof({ namada: null, keplr: null, metamask: null }),
+      url: t.string,
+    }),
+    id: t.keyof({
+      namada: null,
+      cosmos: null,
+      ethereum: null,
+    }),
+    rpc: t.string,
+  }),
+  t.partial({
+    ibc: t.type({
+      portId: t.string,
+    }),
+  }),
+]);
+type ChainType = t.TypeOf<typeof Chain>;
```

Comment:
It would be nice to think it through though. This type is inferred from schema and probably schema should not be a part of types, plus we most likely do not want to always import type from schema files as this would make dependencies all weird. Maybe redundancy is a good thing here. Also it might be the case that types are "accidentaly" the same? :D So in the future maybe those two diverge? WDYT?

Reply:
Yeah maybe that is better actually. I guess since the service methods take e.g. `Chain` and then pass it to a function with parameter `ChainType` it is still getting type checked. And maybe it is actually better not to couple the stored type to the type used everywhere else, since everywhere else shouldn't care how the data is stored.